src/Chapter1.hs

```diff
@@ -542,7 +543,14 @@ value after "=" where the condition is true.
 Casual reminder about adding top-level type signatures for all functions :)
 -}
 
-mid x y z = error "mid: not implemented!"
+mid :: Ord a => a -> a -> a -> a
+mid x y z
+        | x <= y && y <= z = y
+        | z <= y && y <= x = y
+        | y <= x && x <= z = x
+        | z <= x && x <= y = x
+        | x <= z && z <= y = z
+        | y <= z && z <= x = z
```

Comment:
As we mentioned, the compiler in Haskell is very attentive to the exhaustive pattern-matching. And here it would warn you that `Pattern matching is not exhaustive`, as the guards have quite complicated logic, and the compiler won't be able to prove that it covers all the cases.

Because of that, you will need to add another guard – `| otherwise`, to tell the compiler, that your pattern matching is exhaustive 🙂
However, in your case, you rightly noticed, that you covered all cases. So actually if the first 4 cases won't fit, then the answer would definitely be `z`. That means that you can move all these 2 last cases under a single otherwise and return z in here.
```suggestion
        | otherwise = z
```